CSE-1409 enable sys acc

voodoods opened the pull request:
**Jira Issue**

https://8select.atlassian.net/browse/CSE-1409

**Summary**
- call `_8select.initCSE()` in cart confirmation modal when SYS-ACC is enabled in plugin config
- added plugin config option to add a custom HTML container for SYS-ACC

**Checklist**

- [x] PR title is prefixed with Jira Issue Id - e.g. CSE-42
- [x] Add feature / bug label
- [ ] Unit tests for new / changed logic exist OR no logic changes are passing

Live-Demo at: http://shopware.staging.8select.io:8080/

**New Config Option:**
<img width="837" alt="bildschirmfoto 2018-11-20 um 12 39 42" src="https://user-images.githubusercontent.com/1707307/48771337-70327b80-ecc1-11e8-93f6-b702f5832117.png">

**Example - HTML-Container headline in PSV**
<img width="529" alt="bildschirmfoto 2018-11-20 um 12 48 37" src="https://user-images.githubusercontent.com/1707307/48771843-c9e77580-ecc2-11e8-81ff-e29c6f46bc80.png">

**Example - HTML-Container headline in SYS-ACC**
<img width="819" alt="bildschirmfoto 2018-11-20 um 12 48 14" src="https://user-images.githubusercontent.com/1707307/48771846-ce139300-ecc2-11e8-9769-f15d2f6378b4.png">

**Example - with SYS-ACC:**
<img width="792" alt="bildschirmfoto 2018-11-20 um 12 35 37" src="https://user-images.githubusercontent.com/1707307/48771345-74f72f80-ecc1-11e8-9af9-c6c18b5343b6.png">

**Example - without SYS-ACC (default Shopware cross-selling):**
<img width="804" alt="bildschirmfoto 2018-11-20 um 12 36 32" src="https://user-images.githubusercontent.com/1707307/48771371-86d8d280-ecc1-11e8-9d78-2dc0d3eaaa34.png">

Wytrykus commented:
The more I think about it, the more I tend to remove the option `8s_sys_acc_replace_sw_cross_selling` and simply always replace the SW CS part - until we found a better solution how to display both in the modal in a proper way.
If no 8select set is found, the SW CS would be shown anyway, right?

voodoods commented:
@Wytrykus I followed your suggestion and removed the config option to optionally replace shopware's default cross selling. Instead we will now replace it by default, if set is available.

Furthermore I've discussed with @akleiber that we want to give the tenant the option to enhance the SYS-ACC HTML container separately, so I've added an config option for that. See also Screenshots.

voodoods commented:
@Wytrykus like this? :)

![bildschirmfoto 2018-11-22 um 11 45 49](https://user-images.githubusercontent.com/1707307/48898124-40fc4580-ee4c-11e8-9d4e-d808ebc2e2f2.png)

akleiber commented on CseEightselectBasic/Resources/views/frontend/checkout/ajax_add_article.tpl:
```diff
@@ -1,10 +1,31 @@
-        <div data-sku="{$sArticle.ordernumber}" data-include-css="true" data-8select-widget-id="sys-acc"></div>
+        
+        <div class="modal--article block-group">
+            <div class="eightselect-sysacc-html" style="display: none">{include file="string:{$htmlSysAccContainer.0}"}</div>
```
this should be named `-eightselect-widget-container` in order to hide the element if no set is found (ok it is hidden initially - but we should stick to our own API)

see https://docs.8select.io/de/javascript-sdk/widgets/erweitert#container-elemente-ueberschriften-erweiterungen

akleiber replied:
this does not work if you look at the generated HTML - because you just have added a div block with opening and closing tag

<img width="778" alt="bildschirmfoto 2018-11-29 um 01 46 39" src="https://user-images.githubusercontent.com/1506973/49191879-c74ae700-f378-11e8-9f88-24a21455fe43.png">

akleiber commented on CseEightselectBasic/Resources/views/frontend/checkout/ajax_add_article.tpl:
```diff
@@ -1,10 +1,31 @@
-    {$smarty.block.parent}
+
+{block name='checkout_ajax_add_cross_selling'}
+    <div class="eightselect-sw-cross-selling-container">
```
why do we add div around the parent block?

voodoods replied:
Because the parent contains the standard shopware cross selling widget, which we want to hide if we can show a set via sys-acc instead, see: https://github.com/8select/shopware-plugin-sob/pull/63/files#diff-7234b6c9286769a1498ae929fc26e100R166

voodoods commented:
@akleiber done, please take another look.

akleiber commented on CseEightselectBasic/Resources/config.xml:
```diff
@@ -84,32 +94,29 @@
-            <label lang="en">Display SYS-ACC Widget</label>
+            <label lang="de">SYS-ACC: Zeige Widget in der Warenkorb-Bestätigung an</label>
+            <label lang="en">SYS-ACC: Show Widget in cart confirmation</label>
             <value>false</value>
```
make the default `true` please

voodoods replied:
done

voodoods replied:
changed it back to `false`, since `true` gets automatically applied on plugin update. this is undesirable until we can't provide a decent styling for sys-acc (see https://github.com/8select/lambda-deployment-service/pull/37)

besides, we don't have a decent way to preserve customer intended settings before applying new defaults yet - meaning: we don't know if a `false` setting was the default setting before the update or an intentional setting by the tenant.

we should discuss and handle this in a separate issue. until then sys-acc has to be activated manually by the tenant.

voodoods commented:
@akleiber 
- Default Werte und Bezeichnungen in Plugin Config angepasst und überprüft
- SysAcc HTML Container Elemente einblenden hatte nichts mit den DIV blocks an sich zu tun sondern  damit dass es mehrere sind und ich immer nur das jeweils erste eingeblendet habe; lag daran dass ich nicht über alle gefundenen div-elemente mit der betroffenen klasse iteriert habe. ist jetzt fixed.
